### ISSUE TITLE
OptionalEquality is not triggered on comparison to null

### ISSUE THREAD

**Issue by dpursehouse** [google/error-prone#1488]
### Description of the problem / feature request:

Code that compares an `Optional` to `null` does not trigger the `OptionalEquality` warning.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Code like this:

```
  if (someObject.getValue() != null) {
    // do something
  }
```

In the case where the `someObject.getValue()` is in an API where the method previously returned a nullable value, but has been migrated to return an `Optional`, this code will not work properly. 

The `!= null` check should have been replaced with `.isPresent()`, but it will still compile without any warnings, and will just not work as expected.  Bugs like this are difficult to track down, so it would be good if error prone could flag it.

### What version of Error Prone are you using?

I believe v2.2.0 as used in Bazel 2.0.0

###  Have you found anything relevant by searching the web?

Looking at error-prone's tests for `OptionalEquality` it seems to be explicitly testing that comparison to null does not raise a warning. There is no test for comparison to not null.

**Comment by cushon** [comment]
`OptionalEquality` is concerned primarily with using reference equality on `non`-null references, this wasn't an oversight. Comparing an `Optional` with `null` isn't necessarily a bug, see e.g.: https://stackoverflow.com/questions/9561295/whats-the-point-of-guavas-optional-class#comment12143090_9561334

**Comment by dpursehouse** [comment]
Fair enough, although I would argue that in many cases it _is_ a bug, particularly in the scenario that I described in the report.

For now I am working around this by relying on Sonar Lint to detect these cases, rather than Error Prone.
